ext/lmdb-safe/lmdb-safe.hh

```diff
@@ -268,62 +277,62 @@ inline string_view MDBOutVal::getNoStripHeader<string_view>() const
 class MDBInVal
 {
 public:
-  MDBInVal(const MDBOutVal& rhs): d_mdbval(rhs.d_mdbval)
+  MDBInVal(const MDBOutVal& rhs) :
+    d_mdbval(rhs.d_mdbval)
   {
   }
 
 #ifndef DNSDIST
-  template <class T,
-            typename std::enable_if<std::is_integral<T>::value,
-                                    T>::type* = nullptr>
-  MDBInVal(T i)
+  template <class T>
+  MDBInVal(T rhs)
   {
-    static_assert(sizeof(T) == 4, "this code currently only supports 32 bit integers");
-    auto j = htonl(i);    // all actual usage in our codebase is 32 bits. If that ever changes, this will break the build and avoid runtime surprises
-    memcpy(&d_memory[0], &j, sizeof(j));
-
-    d_mdbval.mv_size = sizeof(T);
-    d_mdbval.mv_data = d_memory;;
+    auto rhsNetworkOrder = hostToNetworkByteOrder(rhs);
+    memcpy(&d_memory[0], &rhsNetworkOrder, sizeof(rhsNetworkOrder));
```

Comment:
this looks like a footgun in case things become bigger than 64 bits at some point. Could probably be checked statically?

Reply:
Do you mean e.g. forgetting to widen `d_memory`?

Reply:
yes